vcd/metadata.go

```diff
@@ -4,9 +4,131 @@ import (
 	"fmt"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
 	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"regexp"
+	"strings"
 )
 
+// ignoreMetadataSchema returns the schema associated to ignore_metadata for the provider configuration.
+func ignoreMetadataSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeSet,
+		Optional:    true,
+		Description: "Defines a set of `metadata_entry` that need to be ignored by this provider. All filters on this attribute are computed with a logical AND",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"resource_type": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Ignores metadata from the specific resource type",
```

Comment:
Would it be beneficial to add a ValidateFunc like StringInSlice here? Or is it not needed because of terraformMetadataRelation?

Reply:
I think it's a good idea, I added the validation (and removed the part in the docs that said that a wrong type would have no effect, as now it will fail)